src/Uno.UWP/Devices/Sensors/Barometer.cs

```diff
@@ -0,0 +1,55 @@
+#if __ANDROID__ || __IOS__
+
+using Windows.Foundation;
+
+namespace Windows.Devices.Sensors
+{
+	public partial class Barometer
+	{
+		private static bool _initializationAttempted = false;
+		private static Barometer _instance = null;
+		private static int _activeSubscribers = 0;
+
+		private TypedEventHandler<Barometer, BarometerReadingChangedEventArgs> _readingChanged;
+
+		/// <summary>
+		/// Hides the public parameterless constructor
+		/// </summary>
+		private Barometer()
+		{
+		}
+
+		public static Barometer GetDefault()
+		{
+			if (_instance == null && !_initializationAttempted)
+			{
+				_instance = TryCreateInstance();
+				_initializationAttempted = true;
+			}
+			return _instance;
+		}
+
+		public event TypedEventHandler<Barometer, BarometerReadingChangedEventArgs> ReadingChanged
```

Comment:
Doesn't seem to be thread-safe. Does UWP permit subscription to this event off of the UI thread?

Reply:
Looks like UWP does permit subscription off the UI thread, see https://github.com/microsoft/Windows-universal-samples/blob/cdfd8ef6b38a710eb5e85c776a48c8369b4c74de/Samples/Accelerometer/cs/Scenario1_DataEvents.xaml.cs#L88

```
ReadingChanged += (s,e) => { 
    await Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
    {
        // Update UI with e.Reading
    });
};  
```

Reply:
You are right, I will add some lock

Reply:
@ghuntley Interesting code snippet, seems like it's actually _raised_ off the UI thread on UWP. (Makes sense)

Thanks @MartinZikmund :)